saleor/graphql/warehouse/mutations.py

```diff
@@ -59,6 +61,62 @@ def prepare_address(cls, cleaned_data, *args):
         return address_form.save()
 
 
+class WarehouseShippingZoneAssign(WarehouseMixin, ModelMutation, I18nMixin):
+    warehouse = graphene.Field(
+        Warehouse, description="A warehouse to add shipping zone."
+    )
+
+    class Meta:
+        model = models.Warehouse
+        permissions = (ProductPermissions.MANAGE_PRODUCTS,)
+        description = "Add shipping zone to given warehouse."
+        error_type_class = WarehouseError
+        error_type_field = "warehouse_errors"
+
+    class Arguments:
+        id = graphene.ID(description="ID of a warehouse to update.", required=True)
+        shipping_zone_ids = graphene.List(
```

Comment:
I think we should use `graphene.NonNull` on `graphene.ID`, to prevent executing mutation on the empty list.